node/src/main/kotlin/net/corda/node/services/network/PersistentNetworkMapCache.kt

```
@@ -99,17 +100,11 @@ open class PersistentNetworkMapCache(cacheFactory: NamedCacheFactory,
 
     override fun getPartyInfo(party: Party): PartyInfo? {
         val nodes = getNodesByLegalIdentityKey(party.owningKey)
-        if (nodes.size == 1 && nodes[0].isLegalIdentity(party)) {
-            return PartyInfo.SingleNode(party, nodes[0].addresses)
+        return when {
+            nodes.isEmpty() -> throw PartyNotFoundException("Could not find party: $party", party.name)
```

Isn't it API break?

---

What do you mean by that?

---

`getPartyInfo()` is a part of public API. The behavior was changed in incompatible way to throw an exception instead of returning null.

---

Yeah, ok, @rekalov is right. Since `getPartyInfo` is public api and we now throw an exception it is possible to _break_ the behaviour of an existing application. We could move the check that throws `PartyNotFoundException` somewhere else where it only affects the state machine rather than the user code in the flow. @rick-r3 thoughts?

---

Good catch.  Maybe we need to track back to the call sites from the messaging layer.  Looks like the call site we care about is `FlowMessagingImpl.createMessage(Destination, SessionMessage, SenderDeduplicationId)  (net.corda.node.services.statemachine)` ?  That has a `requireNotNull` where we can throw this exception instead.

Worth a scan of Enterprise code base to see if we access `getPartyInfo` method when creating messages differently anywhere there that would now need to be changed also.

I think it is worth keeping the new Exception in public though.  Technically it is a small semantic change but Cordapps shouldn't intentionally be probing the network map via message sends.

---

I didn't see any additional uses in Enterprise, so perhaps that simple.